pulsar/internal/connection.go

```diff
@@ -546,8 +555,13 @@ func (c *connection) Write(data Buffer) {
 
 func (c *connection) SendRequest(requestID uint64, req *pb.BaseCommand,
 	callback func(command *pb.BaseCommand, err error)) {
-	if c.getState() == connectionClosed {
+	c.incomingRequestsWG.Add(1)
```

Comment:
Why is this waitgroup needed for both send requests?

Reply:
As mentioned https://github.com/apache/pulsar-client-go/pull/539#discussion_r650127814 we need to track all writes to the incomingRequestsCh